Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/Configuration/BlobStorageConfigurationItem.cs

```diff
@@ -0,0 +1,35 @@
+﻿// <copyright file="BlobStorageConfigurationItem.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.TenantManagement.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using Corvus.Azure.Storage.Tenancy;
+
+    /// <summary>
+    /// Enrollment configuration item for tenanted blob storage config.
+    /// </summary>
+    public class BlobStorageConfigurationItem : StorageConfigurationItem<BlobStorageConfiguration>
+    {
+        /// <summary>
+        /// The content type of the configuration item.
+        /// </summary>
+        public const string RegisteredContentType = ConfigurationItem.BaseContentType + "azureblobstorage";
+
+        /// <inheritdoc/>
+        public override string ContentType => RegisteredContentType;
+
+        /// <inheritdoc/>
+        public override IEnumerable<KeyValuePair<string, object>> AddConfiguration(IEnumerable<KeyValuePair<string, object>> values)
+        {
+            if (this.Configuration.Container == null)
+            {
+                throw new NullReferenceException($"{nameof(this.Configuration.Container)} cannot be null.");
```

Comment:
It's valid for `Container` to be null. If not provided, we fall back to that provided in `BlobStorageConfiguration` - see https://github.com/corvus-dotnet/Corvus.Tenancy/blob/2c49f150c9fa093b7ac983175694f16d3436061a/Solutions/Corvus.Azure.Storage.Tenancy/Corvus/Azure/Storage/Tenancy/Internal/TenantCloudBlobContainerFactory.cs#L189